### PR TITLE
refactor: duplication

### PR DESCRIPTION
**Pull request by yudukikun5120**

### CODE REVIEW

File: src/methods/filterCourses.ts [hunk 1]
```diff
@@ -135,27 +121,22 @@ const createModuleSelector = () => {
...
+  if (
+    !(coursesListContainer || coursesThumbnailContainer) ||
+    (coursesListContainer && coursesThumbnailContainer)
   )
+    return
 
-  if (coursesListContainer) {
-    viewMode = "list"
+  const coursesContainer = coursesListContainer ?? coursesThumbnailContainer
+  if (!coursesContainer) return
```
Comment (yudukikun5120) [inline src/methods/filterCourses.ts:137]: @mkobayashime I think the last return expression is useless, but TSC says coursesContainer can be null. Why does it happen?

Comment (yudukikun5120) [reply src/methods/filterCourses.ts]: @mkobayashime I got the answer. Thank you.

**Comment by yudukikun5120** [comment]
@mkobayashime Could you review this PR?

File: src/methods/filterCourses.ts [hunk 2]
```diff
@@ -135,27 +121,17 @@ const createModuleSelector = () => {
...
-  let courses: HTMLElement[]
-
-  const coursesListContainer = document.querySelector(".courselist tbody")
-  const coursesThumbnailContainer = document.querySelector(
-    ".mycourses-body .section"
-  )
+  const [coursesListContainer, coursesThumbnailContainer] = [
+    document.querySelector<HTMLTableElement>(".courselist tbody"),
+    document.querySelector<HTMLElement>(".mycourses-body .section"),
+  ]
```
Comment (mkobayashime) [inline src/methods/filterCourses.ts:127]:
```suggestion
  const coursesListContainer =
    document.querySelector<HTMLElement>(".courselist tbody")
  const coursesThumbnailContainer = document.querySelector<HTMLElement>(
    ".mycourses-body .section"
  )
```

I don't see any positive reason for using destructuring assignments here. Do you hava some?

File: src/methods/filterCourses.ts [hunk 3]
```diff
@@ -282,3 +258,18 @@ const applyFilter = (moduleCode: ModuleCode): void => {
     })
   }
 }
+
+const getCourses = (
+  coursesContainer: HTMLTableElement | HTMLElement,
```
Comment (mkobayashime) [inline src/methods/filterCourses.ts:263]:
```suggestion
  coursesContainer: HTMLElement,
```

File: src/methods/filterCourses.ts [hunk 4]
```diff
@@ -282,3 +258,18 @@ const applyFilter = (moduleCode: ModuleCode): void => {
...
+  viewMode: string
+) =>
+  [...Array(coursesContainer.childElementCount)]
+    .map((_, i) => coursesContainer.children[i])
+    .filter(
+      (_, i) =>
+        i !==
+        rowIndexToBeExcluded(viewMode, coursesContainer.childElementCount - 1)
+    ) as HTMLElement[]
```
Comment (mkobayashime) [inline src/methods/filterCourses.ts:272]:
```suggestion
  (Array.from(coursesContainer.children) as HTMLElement[]).filter(
    (_, i) =>
      i !==
      rowIndexToBeExcluded(viewMode, coursesContainer.childElementCount - 1)
  )
```

**Review by mkobayashime — CHANGES_REQUESTED** [review]
Thanks for the work!

**Comment by mkobayashime** [comment]
@yudukikun5120 
Could you run Prettier and squash commits into one?

**Comment by yudukikun5120** [comment]
@mkobayashime May I squash the commits when merged?

**Comment by mkobayashime** [comment]
@yudukikun5120 
Sorry for the confusion. I'd like you to squash these 5 (previously 4) commits into 1 commit, not the merge with squash.

**Review by mkobayashime — APPROVED** [review]
Thanks!

**Comment by yudukikun5120** [comment]
@mkobayashime BTW, what do we have to squash the commits before the PRs merge for? What do squashing before the merges and "squash and merge" differ?

**Comment by mkobayashime** [comment]
@yudukikun5120 
Sorry, I actually confused _Squash and merge_ with _Rebase and merge_. I'll welcome _Squash and merge_ from next time, as long as:
- commits can be bundled into 1 commit semantically (don't have multiple intention)
- commit message is properly set for the new (squashed) commit

**Pull request merged**